app/recipes/[id]/RecipeDetailClient.tsx

```diff
@@ -0,0 +1,244 @@
+                      }
+                    >
+                      {index + 1}
+                    </div>
+                    <p className="flex items-center leading-relaxed lg:text-lg flex-1 text-[#374151]">
+                      {step}
+                    </p>
+                  </div>
+                ))}
+              </div>
+            </div>
+          </div>
+
+          {/* 재료 및 액션 사이드바 */}
+          <div className="lg:col-span-1 space-y-6">
+            {/* 필요한 재료 */}
+            <div className="lg:col-span-1 space-y-6">
+              <div className="bg-white rounded-2xl p-6 shadow-sm border border-[#E5E7EB]">
+                <h2 className="lg:text-2xl font-bold text-[#374151] mb-4 flex items-center gap-2">
+                  <div className="p-2 bg-[#10B981]/10 rounded-xl">
+                    <LeafyGreen className="w-6 h-6 text-[#10B981]" />
+                  </div>
+                  필요한 재료
+                </h2>
+                <div className="space-y-3">
+                  {recipe.ingredients.map(({ name, quantity }, _) => {
+                    const isMissing = missingIngredients.includes(name);
+                    const available = !isMissing;
+                    const userIngredient = userIngredientList.find(
+                      (userIng) =>
+                        userIng.name.toLowerCase() === name.toLowerCase()
+                    );
+
+                    return (
+                      <div
+                        key={name}
+                        className={`flex items-center justify-between p-3 rounded-xl ${
+                          available
+                            ? "bg-[#F0FDF4] border border-[#10B981]/20"
+                            : "bg-[#FEF2F2] border border-[#EF4444]/20"
+                        }`}
+                      >
+                        <div className="flex items-center gap-3">
+                          {available ? (
+                            <CheckCircle className="w-5 h-5 text-[#10B981]" />
+                          ) : (
+                            <XCircle className="w-5 h-5 text-[#EF4444]" />
+                          )}
+                          <span
+                            className={`font-medium ${
+                              available ? "text-[#047857]" : "text-[#DC2626]"
+                            }`}
+                          >
+                            {name}
+                          </span>
+                          {available &&
+                            userIngredient &&
+                            userIngredient.quantity > 0 && (
+                              <span className="text-xs text-[#047857] bg-[#DCFCE7] px-2 py-1 rounded-full">
+                                보유: {userIngredient.quantity}개
+                              </span>
+                            )}
```

Comment:
```suggestion
                          {hasUserIngredientQuantity && (
                              <span className="text-xs text-[#047857] bg-[#DCFCE7] px-2 py-1 rounded-full">
                                보유: {userIngredient.quantity}개
                              </span>
                            )}
```

`hasUserIngredientQuantity`라는 변수를 만들어서 사용해주면 좋을 것 같아요. 리뷰어 입장에서는 hasUserIngredientQuantity를 보고 갯수가 있으면 보여준다고 생각할 수 있을 것 같아요